pallets/funding/src/tests/3_auction.rs

```diff
@@ -1,1300 +1,1699 @@
 
-#[test]
-fn pallet_can_start_auction_automatically() {
-	let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
-	let project_id = inst.create_evaluating_project(default_project_metadata(ISSUER_1), ISSUER_1);
-	let evaluations = default_evaluations();
-	let required_plmc = MockInstantiator::calculate_evaluation_plmc_spent(evaluations.clone());
-	let ed_plmc = required_plmc.accounts().existential_deposits();
-
-	inst.mint_plmc_to(required_plmc);
-	inst.mint_plmc_to(ed_plmc);
-	inst.evaluate_for_users(project_id, evaluations).unwrap();
-	inst.advance_time(<TestRuntime as Config>::EvaluationDuration::get() + 1).unwrap();
-	assert_eq!(inst.get_project_details(project_id).status, ProjectStatus::AuctionInitializePeriod);
-	inst.advance_time(<TestRuntime as Config>::AuctionInitializePeriodDuration::get() + 2).unwrap();
-	assert_eq!(inst.get_project_details(project_id).status, ProjectStatus::AuctionOpening);
-}
+			assert_ok!(inst.bid_for_users(project_id_usdt, vec![usdt_bid.clone()]));
+			assert_err!(
+				inst.bid_for_users(project_id_usdt, vec![usdc_bid.clone()]),
+				Error::<TestRuntime>::FundingAssetNotAccepted
+			);
+			assert_err!(
+				inst.bid_for_users(project_id_usdt, vec![dot_bid.clone()]),
+				Error::<TestRuntime>::FundingAssetNotAccepted
+			);
 
-#[test]
-fn issuer_can_start_auction_manually() {
-	let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
-	let project_id = inst.create_evaluating_project(default_project_metadata(ISSUER_1), ISSUER_1);
-	let evaluations = default_evaluations();
-	let required_plmc = MockInstantiator::calculate_evaluation_plmc_spent(evaluations.clone());
-	let ed_plmc = required_plmc.accounts().existential_deposits();
-	inst.mint_plmc_to(required_plmc);
-	inst.mint_plmc_to(ed_plmc);
-	inst.evaluate_for_users(project_id, evaluations).unwrap();
-	inst.advance_time(<TestRuntime as Config>::EvaluationDuration::get() + 1).unwrap();
-	assert_eq!(inst.get_project_details(project_id).status, ProjectStatus::AuctionInitializePeriod);
-	inst.advance_time(1).unwrap();
-	inst.execute(|| Pallet::<TestRuntime>::do_auction_opening(ISSUER_1, project_id)).unwrap();
-	assert_eq!(inst.get_project_details(project_id).status, ProjectStatus::AuctionOpening);
-}
+			assert_err!(
+				inst.bid_for_users(project_id_usdc, vec![usdt_bid.clone()]),
+				Error::<TestRuntime>::FundingAssetNotAccepted
+			);
+			assert_ok!(inst.bid_for_users(project_id_usdc, vec![usdc_bid.clone()]));
+			assert_err!(
+				inst.bid_for_users(project_id_usdc, vec![dot_bid.clone()]),
+				Error::<TestRuntime>::FundingAssetNotAccepted
+			);
+
+			assert_err!(
+				inst.bid_for_users(project_id_dot, vec![usdt_bid.clone()]),
+				Error::<TestRuntime>::FundingAssetNotAccepted
+			);
+			assert_err!(
+				inst.bid_for_users(project_id_dot, vec![usdc_bid.clone()]),
+				Error::<TestRuntime>::FundingAssetNotAccepted
+			);
+			assert_ok!(inst.bid_for_users(project_id_dot, vec![dot_bid.clone()]));
+		}
+
+		fn test_bid_setup(
```

Comment:
We can do the actual bid here as well and return the Dispatch result of the bid. Then our test only requires assert_ok! or assert_noop!.

Reply:
`assert_noop` checks that storage wasn't modified, so you need to actually call the extrinsic inside it. Therefore putting the `DispatchResult` wouldn't work

Reply:
Ok I switched to using assert_err and did your suggestion. Since we are calling an extrinsic directly I think we can be sure storage would be rolled back